haproxy.cfg

```diff
@@ -53,32 +56,35 @@ frontend http-in
   use_backend cookie_strategy if routing_strategy_cookie is_new_cookie
 
 backend old_domain
-  http-response set-header X-Proxy-Flow route-to-legacy
-  option tcp-check
   balance leastconn
+  http-response set-header X-Proxy-Flow route-to-legacy
+  http-request set-header Host "$OLD_DOMAIN" if { env(REWRITE_HOST_OLD) -m bool }
 
-  server-template default_old "$SERVER_COUNT" "$OLD_DOMAIN"
+  server-template default_old "$SERVER_COUNT" "$OLD_DOMAIN" "$SERVER_PARAMS_OLD"
 
 backend percentage_strategy
   http-response set-header X-Proxy-Flow route-to-percentage
-  option tcp-check
   # https://www.haproxy.com/documentation/haproxy-configuration-manual/latest/#4.2-balance
   balance static-rr
 
   cookie "$COOKIE_PERCENTAGE_NAME" insert indirect
 
+  # and the weight juggeling picked old or new domain
+  http-request set-header Host "$NEW_DOMAIN" if { env(REWRITE_HOST_NEW) -m bool && srv_id(new_domain_percentage) }
+  http-request set-header Host "$OLD_DOMAIN" if { env(REWRITE_HOST_OLD) -m bool && srv_id(old_domain_percentage) }
+
   # old domain
-  server-template old_domain_percentage "$SERVER_COUNT" "$OLD_DOMAIN" cookie "old_domain_$PERCENTAGE_OLD" weight "$PERCENTAGE_OLD"
+  server-template old_domain_percentage "$SERVER_COUNT" "$OLD_DOMAIN" cookie "old_domain_$PERCENTAGE_OLD" weight "$PERCENTAGE_OLD" "$SERVER_PARAMS_OLD"
 
   # new domain
-  server-template new_domain_percentage "$SERVER_COUNT" "$NEW_DOMAIN" cookie "new_domain_$PERCENTAGE_NEW" weight "$PERCENTAGE_NEW"
+  server-template new_domain_percentage "$SERVER_COUNT" "$NEW_DOMAIN" cookie "new_domain_$PERCENTAGE_NEW" weight "$PERCENTAGE_NEW" "$SERVER_PARAMS_NEW"
 
 backend cookie_strategy
   balance leastconn
   http-response set-header X-Proxy-Flow route-to-new
-  option tcp-check
```

Comment:
Why is tcp-check removed? Is this the health check?

Reply:
Ok, didn't make sense before, so we just remove it.

Reply:
no the healthcheck is a different option on the server itself called `check`